kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/utils/OpenIDConnectionUtils.java

```diff
@@ -102,11 +102,13 @@ public static CompletableFuture<String> resolveOIDCTokenFromAuthConfig(Config cu
               return accessToken;
             }
 
-            // Persist new config and if successful, update the in memory config.
             try {
+              //update in memory config
+              updateInMemoryConfigWithUpdatedToken(currentConfig, map);
```

Comment:
Given the previous comment, I think that the right place to do this is within the `OpenIDConnectionUtils.persistKubeConfigWithUpdatedAuthInfo`

The method should be refactored a little to have a clear structure of what's bring updated (the file or the in-memory object)

Reply:
ok, make sense